draft-ietf-masque-connect-udp-listen.md

```diff
@@ -94,41 +94,55 @@ Proxying HTTP request.
 {::boilerplate bcp14-tagged}
 
 This document uses terminology from {{CONNECT-UDP}} and notational conventions
-from {{!QUIC=RFC9000}}. This document uses the terms Integer and List from
+from {{!QUIC=RFC9000}}. This document uses the terms Integer, Boolean and List from
 {{Section 3 of !STRUCTURED-FIELDS=RFC8941}} to specify syntax and parsing.
 
 
 # Proxied UDP Binding Mechanism {#mechanism}
 
 In unextended UDP Proxying requests, the target host is encoded in the HTTP
-request path or query. For Bound UDP Proxying, it is instead conveyed in each
-HTTP Datagram, see {{format}}.
+request path or query. For Bound UDP Proxying, the target is either conveyed in each
+HTTP Datagram (see {{format}}), or registered via capsules and then compressed
+(see {{contextid}}).
 
 When performing URI Template Expansion of the UDP Proxying template (see
 {{Section 3 of CONNECT-UDP}}), the client sets both the target_host and the
 target_port variables to the '*' character (ASCII character 0x2A).
 
-Before sending its UDP Proxying request to the proxy, the client allocates an
-even-numbered context ID, see {{Section 4 of CONNECT-UDP}}. The client then adds
-the "Connect-UDP-Bind" header field to its UDP Proxying request, with its
-value set as the allocated context ID, see {{hdr}}.
+When sending the UDP Proxying request to the proxy, the client adds
```

Comment:
Does it need to have "Connect-UDP-Bind" in the response header? Is the following status sufficient, instead?
```
HEADERS
 :status = 200
 capsule-protocol = ?1
```

Reply:
That's not sufficient because it doesn't allow the client to differentiate between a server that knows this extension and supports it vs an old server that's never heard of this extension and completely ignored the Connect-UDP-Bind header. Echoing is a common tool for determining server support